draft-barnes-acme.md

```diff
@@ -129,65 +128,80 @@ In the discussion below, we will refer to three different types of keys / key pa
 Subject Public Key:
 : A public key to be included in a certificate.
 
-Authorized Key Pair:
+Account Key Pair:
 : A key pair for which the ACME server considers the holder of the private key authorized to manage certificates for a given identifier.  The same key pair may be authorized for multiple identifiers.
 
 Recovery Token:
-: A secret value that can be used to demonstrate prior authorization for an identifier, in a situation where all Subject Private Keys and Authorized Keys are lost.
+: A secret value that can be used to demonstrate prior authorization for an identifier, in a situation where all Subject Private Keys and Account Keys are lost.
 
 ACME messaging is based on HTTPS {{RFC2818}} and JSON {{RFC7159}}.  Since JSON is a text-based format, binary fields are Base64-encoded.  For Base64 encoding, we use the variant defined in {{I-D.ietf-jose-json-web-signature}}.  The important features of this encoding are (1) that it uses the URL-safe character set, and (2) that "=" padding characters are stripped.
 
+Some HTTPS bodies in ACME are authenticated and integrity-protected by being encapsulated in a JSON Web Signature (JWS) object {{I-D.ietf-jose-json-web-signature}}.  ACME uses a profile of JWS, with the following restrictions:
+
+* The JWS MUST use the Flattened JSON Serialization
```

Comment:
Why?  What is the reason behind this?

I consider this to be an unreasonable imposition, for the following reason:

Suppose a certain JOSE library serialises a JWS with a single signature
using the General JWS JSON Serialization Syntax.  That library cannot
be used with ACME.

According to the JWS draft there is no requirement for an implementation
to serialise a single-signature JWS using the Flattened syntax, nor is there
a requirement to provide users a knob to control how to deal with this
ambiguous case.

So if there is no important reason to have this requirement in ACME,
please remove it.